Lib/test/test_pep646_syntax.py

```diff
@@ -0,0 +1,268 @@
+doctests = """
+
+Setup
+
+    >>> class AClass:
+    ...    def __init__(self):
+    ...        self._setitem_name = None
+    ...        self._setitem_val = None
+    ...    def __setitem__(self, name, val):
+    ...        self._setitem_name = name
+    ...        self._setitem_val = val
+    ...    def __repr__(self):
+    ...        return f"A[{self._setitem_name}]={self._setitem_val}"
+    ...    def __getitem__(self, name):
+    ...        return ParameterisedA(name)
+    >>> class ParameterisedA:
+    ...    def __init__(self, name):
+    ...        self._name = name
+    ...    def __repr__(self):
+    ...        return f"A[{self._name}]"
+    ...    def __iter__(self):
+    ...        for p in self._name:
+    ...            yield p
+    >>> class B:
+    ...    def __iter__(self):
+    ...        yield StarredB()
+    ...    def __repr__(self):
+    ...        return "B"
+    >>> class StarredB:
+    ...    def __repr__(self):
+    ...        return "StarredB"
+    >>> A = AClass()
+    >>> b = B()
+
+Slices that are supposed to work, starring our custom B class
+
+    >>> A[*b]
+    A[(StarredB,)]
+    >>> A[*b, *b]
+    A[(StarredB, StarredB)]
+    >>> A[b, *b]
+    A[(B, StarredB)]
+    >>> A[*b, b]
+    A[(StarredB, B)]
+    >>> A[b, b, *b]
+    A[(B, B, StarredB)]
+    >>> A[*b, b, b]
+    A[(StarredB, B, B)]
+    >>> A[b, *b, b]
+    A[(B, StarredB, B)]
+    >>> A[b, b, *b, b]
+    A[(B, B, StarredB, B)]
+    >>> A[b, *b, b, b]
+    A[(B, StarredB, B, B)]
+    >>> A[A[b, *b, b]]
+    A[A[(B, StarredB, B)]]
+    >>> A[*A[b, *b, b]]
+    A[(B, StarredB, B)]
+    >>> A[b, ...]
+    A[(B, Ellipsis)]
+    >>> A[*A[b, ...]]
+    A[(B, Ellipsis)]
+
+Slices assignments that are supposed to work, starring our custom B class
```

Comment:
Is it worth also testing `del A[*b]`? That's yet another separate piece of grammar.

Reply:
Oh, great catch! Done. (I've also changed the layout of the test cases so it's easier to be sure we're testing getting, setting and deleting for each possible slice.)